addon/hifi-connections/dummy-connection.js

```diff
@@ -1,74 +1,94 @@
 import Ember from 'ember';
-import DebugLogging from 'ember-hifi/mixins/debug-logging';
+import BaseSound from './base';
 
 let ClassMethods = Ember.Mixin.create({
   setup() {},
   canPlay: () => true,
   canUseConnection: () => true,
   canPlayMimeType: () => true,
+  toString() {
+    return 'Dummy Connection';
+  }
 });
 
-let DummyConnection = Ember.Object.extend(Ember.Evented, DebugLogging, {
+
+let DummyConnection = BaseSound.extend({
   debugName: 'dummyConnection',
-  position: 0,
-  duration: 10000,
-  
-  init() {
-    this.on('audio-played',    () => {
-      this.set('hasPlayed', true);
-      this.set('isLoading', false);
-      this.set('isPlaying', true);
-      this.set('error', null);
-      
-      // recover lost isLoading update
-      this.notifyPropertyChange('isLoading');
-    });
+  _position: 0,
+  _tickInterval: 50,
+  setup() {
+    let {result} = this.getInfoFromUrl();
+    if (result === 'bad') {
+      Ember.run.next(() => this.trigger('audio-load-error', this));
+    }
+    else {
+      Ember.run.next(() => this.trigger('audio-ready', this));
+    }
+  },
 
-    this.on('audio-paused',   () => {
-      this.set('isPlaying', false);
-    });
-    this.on('audio-ended',    () => { 
-      this.set('isPlaying', false);
-    });
-    
-    this.on('audio-load-error', (e) => {
-      if (this.get('hasPlayed')) {
-        this.set('isLoading', false);
-        this.set('isPlaying', false);
-      }
-      this.set('error', e);
-    });
+  stopTicking: function() {
+    window.clearTimeout(this.tick);
+  },
 
-    this.on('audio-loaded', () => {
-      this.set('isLoading', false);
-    });
-    
-    Ember.run.next(() => this.trigger('audio-ready'));
+  startTicking: function() {
+    this.tick = window.setTimeout(Ember.run.bind(() => {
+      this._setPosition((this._currentPosition() || 0) + this.get('_tickInterval'));
+      this.startTicking();
+    }), this.get('_tickInterval'))
   },
+
+  getInfoFromUrl: function() {
+    let [, result, length, name] = this.get('url').split('/');
```

Comment:
it'd be nice if there was some error or syntax checking here in case someone uses this in a test and doesn't quit get it.